### PR TITLE
cluster up: optionally install metrics components

### PR DESCRIPTION
**Pull request by csrwng**
Adds an experimental flag to `oc cluster up` to install metrics components.

When specifying the --metrics flag, the master asset config is updated with the hostname for the metrics route and the pod to deploy metrics components is created.

### CODE REVIEW

**Comment by csrwng** [comment]
@smarterclayton ptal
/cc @jwforres @spadgett

**Comment by csrwng** [comment]
Opened the following issues on origin-metrics:
https://github.com/openshift/origin-metrics/issues/148
https://github.com/openshift/origin-metrics/issues/149

File: pkg/bootstrap/docker/openshift/metrics.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+					VolumeSource: kapi.VolumeSource{
+						Secret: &kapi.SecretVolumeSource{
+							SecretName: "metrics-deployer",
+						},
+					},
+				},
+			},
+		},
+	}
+	pod.GenerateName = "metrics-deployer-"
```
Comment (smarterclayton) [inline pkg/bootstrap/docker/openshift/metrics.go]: Why are you using a generate name?

Comment (csrwng) [inline pkg/bootstrap/docker/openshift/metrics.go]: Hadn't given it much thought. The original template has a generate name. But in my case, I guess it doesn't make much sense. I'll switch it to a regular name.

**Comment by smarterclayton** [comment]
No other comments beyond that.

**Comment by csrwng** [comment]
Comment addressed. 

Also added a line with the metrics URL in the info part when you specify the --metrics flag:

```
-- Server Information ...
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.56.1:8443

   The metrics service is available at:
       https://metrics-openshift-infra.192.168.56.1.xip.io

   You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin
```

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**